nifti2/nifti2_io.c

```diff
@@ -5021,7 +5022,7 @@ nifti_image* nifti_convert_n2hdr2nim(nifti_2_header nhdr, const char * fname)
 
    nim->nifti_type = (is_onefile) ? NIFTI_FTYPE_NIFTI2_1 : NIFTI_FTYPE_NIFTI2_2;
 
-   int byteOrder = nifti_short_order() ;
+   byteOrder = nifti_short_order() ;
```

Comment:
Why split the declaration and initial assignment?  The codebase clearly doesn't support C89, as things like `int64_t` are used unconditionally.

Reply:
While not required by the compiler, it is certainly the convention in the code, making location of types predictable.

Reply:
OK, if that's the project's style, I can follow it in the future, and update my other pending PRs accordingly.

But I don't think that style is advantageous.  It's not for nothing that C99 and C++ got rid of the old C requirement for having declarations at the top.  It puts everything in a larger scope than necessary, see for example:

https://wiki.sei.cmu.edu/confluence/display/c/DCL19-C.+Minimize+the+scope+of+variables+and+functions

Tools like cppcheck and clang-tidy can warn about such things. I was going to work on such a scope reduction patch after my other cppcheck PR lands.

Reply:
Yes, please do not alter the style of programming in this library.  Save that for a new library if you wish to.

Reply:
OK, will keep the C89 style in future changes.